activesupport/lib/active_support/core_ext/file/lock.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class File
```

Comment:
why you monkey patching the File class ?

Reply:
This is a feature that should be available outside of `FileStore`. `ActiveSupport` is made of monkey patches. Just see `atomic_write` for example.